crates/torin/src/values/size.rs

```diff
@@ -203,28 +215,283 @@ impl std::fmt::Display for DynamicCalculation {
 struct DynamicCalculationEvaluator<'a> {
     calcs: Iter<'a, DynamicCalculation>,
     parent_value: f32,
+    scaling_factor: f32,
     root_value: f32,
     current: Option<&'a DynamicCalculation>,
 }
 
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum Val {
+    Scaled(f32),
+    UnScaled(f32),
+    PartiallyScaled { scaled: f32, unscaled: f32 },
```

Comment:
Can you explain what each variant is for? And also explain a bit the custom SubAssign, AddAssign, etc... impls?

Reply:
I mean as a comment